### PR TITLE
Env2mfile changes needed for Debian packaging

### PR DESCRIPTION
**Pull request by jpakkane**
Do not merge yet. This needs some coordination with Debian.

### CODE REVIEW

File: mesonbuild/scripts/env2mfile.py [hunk 1]
```diff
@@ -131,21 +131,26 @@ def get_args_from_envvars(infos: MachineInfo) -> None:
...
     'mips64el': 'mips64',
     'i686': 'x86',
 }
-cpu_map = {
+
+deb_cpu_map = {
     'armhf': 'arm7hlf',
-    'mips64el': 'mips64'
+    'mips64el': 'mips64',
+    'powerpc64le': "ppc64",
```
Comment (eli-schwartz) [inline mesonbuild/scripts/env2mfile.py:142]:
```suggestion
    'powerpc64le': 'ppc64',
```

File: mesonbuild/scripts/env2mfile.py [hunk 2]
```diff
@@ -160,7 +160,7 @@ def deb_compiler_lookup(infos: MachineInfo, compilerstems: T.List[T.Tuple[str, s
             pass
 
 def detect_cross_debianlike(options: T.Any) -> MachineInfo:
-    if options.debarch is None:
+    if options.debarch == 'auto':
```
Comment (eli-schwartz) [inline mesonbuild/scripts/env2mfile.py:168]: "add autodetect" -- more like fix the current broken one. :D Looks like we only evaluate this function if options.debarch is not None, and then the first thing we do is check if it is None. Which cannot ever be the case.

**Review by eli-schwartz — COMMENTED** [review]
> This needs some coordination with Debian.

All of these other than the ppc64 addition appear to be simple coding errors or refactoring with no runtime effect. Anything in particular that we are waiting on?

**Comment by helmutg** [comment]
LGTM (from a Debian pov)

**Comment by jpakkane** [comment]
I'm specifically _not_ tagging this for 1.0.2 because this code will only be used after Bookworm. So even if a 1.0.2 happens before feature freeze, we want to keep the old functionality unchanged to minimize regression possibilities.

**Pull request merged**